### PR TITLE
Add support for condensing full.jsonl -> transcript.jsonl (opencode and claude first)

### PR DESCRIPTION
**Pull request by computermode**
Pulling out changes that have been tested from https://github.com/entireio/cli/pull/754 to keep the diffs smaller.

With these changes, Opencode and Claude `full.jsonl` files can be transformed to a uniform `transcript.jsonl` format that only includes what the UI needs. The changes have also been validated with the UI: the generated `transcript.jsonl` format is parse-able for both of these agent types.

You may notice that the Claude changes are in a `compact.go` file: that's because that file will include what all the agents have in common. Since Claude is the default and has a decent amount of crossover already with its `full.jsonl` with other agents, that's why its code is within that shared file. The Opencode agent parser uses some of the helper methods from it.

Below are screenshots from the UI where I supplied the code with real `full.jsonl` files and generated `transcript.jsonl` files that I manually pushed up to the checkpoints v1 branch on a test repo (the `metadata.json` had to be updated to route to `transcript.jsonl` as well of course).

Screenshot 2026-03-25 at 3 48 49 PM

Screenshot 2026-03-26 at 4 43 01 PM

---

> [!NOTE]
> **Medium Risk**
> Introduces new transcript-compaction logic that changes how tool results and streaming assistant fragments are merged/omitted, which could affect downstream UI parsing across multiple agent formats.
> 
> **Overview**
> Adds a new `transcript/compact` package that converts agent transcripts into a normalized `transcript.jsonl` format with stable field ordering and per-line metadata (`v`, `agent`, `cli_version`).
> 
> For JSONL-based agents (Claude Code/Cursor), the converter now **filters non-semantic event types**, strips assistant thinking/redacted blocks and extra tool fields, **merges streaming assistant fragments by message ID**, and **inlines `tool_result` outputs into preceding `tool_use` blocks** while dropping tool-result-only user lines.
> 
> Adds OpenCode support by detecting the single-JSON session export format and emitting equivalent compact user/assistant lines (including tool parts and timestamps), plus comprehensive fixtures and unit tests covering truncation, mixed formats, and edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit f672fc924123171f9b2ec279d9deeffecedc88cd. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: cmd/entire/cli/transcript/compact/opencode.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+		"id", mustMarshal(msg.Info.ID),
+		"content", json.RawMessage(contentJSON),
+	)
+	return [][]byte{b}
+}
+
+// openCodeToolResult builds the compact {"output":"...","status":"done"|"error"}
+// object from an OpenCode tool state map.
+func openCodeToolResult(state map[string]json.RawMessage) json.RawMessage {
+	status := "done"
```
Comment (Soph) [inline cmd/entire/cli/transcript/compact/opencode.go]: https://github.com/entireio/cli/pull/788/changes#diff-fdfe81444acf8eb918c765c7bae4586d75c8e99ac453cfe2270742d21188cd00R350

uses "success" 🤔

File: cmd/entire/cli/transcript/compact/testdata/claude_full.jsonl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+{"parentUuid":"d6a5b6bf-7d8b-4447-8883-5c50e7ba3f48","isSidechain":false,"message":{"model":"claude-opus-4-6","id":"msg_01XBRjv2YCFQs34eaLK7w9XF","type":"message","role":"assistant","content":[{"type":"text","text":"\n\nLet me check the current full commit message first."}],"stop_reason":null,"stop_sequence":null,"usage":{"input_tokens":3,"cache_creation_input_tokens":126,"cache_read_input_tokens":20544,"cache_creation":{"ephemeral_5m_input_tokens":0,"ephemeral_1h_input_tokens":126},"output_tokens":2,"service_tier":"standard","inference_geo":"not_available"}},"requestId":"req_011CZ9ZvfDBgFqpTWiCHR9TB","type":"assistant","uuid":"482ef0c8-cf92-4dc5-b4ea-969fdb72f344","timestamp":"2026-03-18T00:05:13.959Z","userType":"external","entrypoint":"cli","cwd":"/Users/private/entire/test-repos/test-repo","sessionId":"8b89dece-0f9b-4965-8436-fcec80f5ea06","version":"2.1.78","gitBranch":"test-gemini"}
```
Comment (Soph) [inline cmd/entire/cli/transcript/compact/testdata/claude_full.jsonl:11]: Thinking: we probably should keep token usages too 🤔

Comment (computermode) [reply cmd/entire/cli/transcript/compact/testdata/claude_full.jsonl:11]: I cut out anything we don't display in the UI.. we don't display per-step token usage (not that I could see anywhere) but we do show the overall token usage, which is in the metadata.json file.

Comment (computermode) [reply cmd/entire/cli/transcript/compact/testdata/claude_full.jsonl:11]: (If I'm missing it, let me know 🙇🏻)

File: cmd/entire/cli/transcript/compact/opencode.go (hunk 1 above)
Comment (computermode) [reply cmd/entire/cli/transcript/compact/opencode.go]: Pushed up a change here 👍🏻

File: cmd/entire/cli/transcript/compact/testdata/claude_full.jsonl (hunk 2 above)
Comment (Soph) [reply cmd/entire/cli/transcript/compact/testdata/claude_full.jsonl:11]: I just feel this might be good for long term storage especially since current calculation has sometimes issues and depends on how we count the steps. But when we keep it, it would allow us to calculate later again.

Comment (computermode) [reply cmd/entire/cli/transcript/compact/testdata/claude_full.jsonl:11]: Makes sense - adding it in! Thank you 😄

File: cmd/entire/cli/transcript/compact/compact.go [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/entireio/cli/cmd/entire/cli/textutil"
+	"github.com/entireio/cli/cmd/entire/cli/transcript"
+)
+
+// Options provides metadata fields written to every output line.
+type Options struct {
```
Comment (pfleidi) [inline cmd/entire/cli/transcript/compact/compact.go]: I wonder if there's a name for this that better matches what it does? `OutputMetadata` or something like that?

File: cmd/entire/cli/transcript/compact/compact.go [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+
+// linePreprocessor transforms a parsed JSONL line before conversion.
+// Used by agent-specific converters to normalize their format (e.g., Droid
+// envelope unwrapping) before the shared pipeline processes it.
+type linePreprocessor func(map[string]json.RawMessage) map[string]json.RawMessage
+
+// compactJSONL converts JSONL transcripts (Claude Code, Cursor) into the
+// compact format.
+func compactJSONL(content []byte, opts Options) ([]byte, error) {
+	return compactJSONLWith(content, opts, nil)
```
Comment (pfleidi) [inline cmd/entire/cli/transcript/compact/compact.go]: Does this indirection need to exist? As far as I can tell, `compactJSONLWith()` could mostly be inlined here, right?
